pkg/serverless/daemon/daemon_test.go

```diff
@@ -82,6 +82,11 @@ func TestTellDaemonRuntimeDoneIfLocalTest(t *testing.T) {
 	request, err := http.NewRequest(http.MethodPost, fmt.Sprintf("http://127.0.0.1:%d/lambda/flush", port), nil)
 	assert.Nil(err)
 	response, err := client.Do(request)
+	if err != nil {
+		// retry once in case the daemon wasn't ready to accept requests
```

Comment:
This test was failing with

```
Error:          Expected nil, but got: &url.Error{Op:"Post", URL:"http://127.0.0.1:52816/lambda/flush", Err:(*net.OpError)(0x1400042e3c0)}```

The message on this error is: `Post "http://127.0.0.1:53303/lambda/flush": dial tcp 127.0.0.1:53303: connect: connection refused`.

I suspected that the daemon just was not ready to accept incoming requests. So, I tested a few things, like adding sleeps and retries. This confirmed my suspicion. 

I like this implementation the best because it only requires the sleep when it's absolutely needed (which is in about 5% of test runs). In which case, we'll just sleep a little bit and try again.

Note the gh action check failures saying "response body must be closed (bodyclose)". I believe we are okay here because in the case where an error is returned from `Client.Do`, the response is always nil and vice versa.